compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -877,9 +877,9 @@ pub struct Place<'tcx> {
 
 #[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
 #[derive(TyEncodable, TyDecodable, HashStable, TypeFoldable, TypeVisitable)]
-pub enum ProjectionElem<V, T> {
+pub enum ProjectionElem<V, T1, T2> {
```

Comment:
At this point it might be worth documenting what these types mean? Their names are not exactly descriptive.